alibi_detect/od/_svm.py

```diff
@@ -0,0 +1,234 @@
+from typing import TYPE_CHECKING, Any, Dict, Optional, Tuple, Union
+
+import numpy as np
+
+from alibi_detect.base import (BaseDetector, FitMixin, ThresholdMixin,
+                               outlier_prediction_dict)
+from alibi_detect.exceptions import _catch_error as catch_error
+from alibi_detect.od.pytorch import SVMTorch
+from alibi_detect.od.sklearn import SVMSklearn
+from alibi_detect.utils._types import Literal
+from alibi_detect.utils.frameworks import BackendValidator
+from alibi_detect.version import __version__
+
+if TYPE_CHECKING:
+    import torch
+
+
+backends = {
+    'pytorch': SVMTorch,
+    'sklearn': SVMSklearn
+}
+
+
+class SVM(BaseDetector, ThresholdMixin, FitMixin):
+    def __init__(
+        self,
+        n_components: Optional[int] = None,
+        backend: Literal['pytorch', 'sklearn'] = 'sklearn',
+        device: Optional[Union[Literal['cuda', 'gpu', 'cpu'], 'torch.device']] = None,
+        kernel: Union['torch.nn.Module', Literal['linear', 'poly', 'rbf', 'sigmoid']] = 'rbf',
+        sigma: Optional[float] = None,
+        kernel_params: Optional[Dict] = None,
+    ) -> None:
+        """Support vector machine (SVM) outlier detector.
+
+        The Support vector machine outlier detector fits a one-class SVM to the reference data.
+
+        Rather than the typical approach of optimizing the exact kernel OCSVM objective through a dual formulation,
```

Comment:
I wonder if this might be a bit too much detail for the highest level docstring. However I do intend to iterate slightly on these docstrings when doing the documentation so happy to leave until then.